packages/e2e/test/local-network/local-network.test.ts

```diff
@@ -110,4 +161,67 @@ describe('Local Network', () => {
     wallet1.shutdown();
     wallet2.shutdown();
   });
+
+  it('will receive rewards for delegated tADA', async () => {
+    // Arrange
+    const amountFromFaucet = 100_000_000_000;
+
+    const wallet1 = await getWallet({ env, logger, name: 'Sending Wallet', polling: { interval: 50 } });
+    const wallet2 = await getWallet({ env, logger, name: 'Receiving Wallet', polling: { interval: 50 } });
+
+    await firstValueFrom(wallet1.wallet.syncStatus.isSettled$.pipe(filter((isSettled) => isSettled)));
+    await firstValueFrom(wallet2.wallet.syncStatus.isSettled$.pipe(filter((isSettled) => isSettled)));
+
+    const [{ address: sendingAddress }] = await firstValueFrom(wallet1.wallet.addresses$);
+
+    const waitForEpoch = async (numberOfEpochsToWait: number) => {
+      const currentEpoch = await firstValueFrom(wallet1.wallet.currentEpoch$);
+      return await firstValueFrom(
+        wallet1.wallet.currentEpoch$.pipe(
+          filter(({ epochNo }) => epochNo > currentEpoch.epochNo + numberOfEpochsToWait)
+        )
+      );
+    };
+
+    logger.debug(`Address ${sendingAddress.toString()} will be funded with ${amountFromFaucet} tLovelace.`);
+    await faucetProvider.request(sendingAddress.toString(), amountFromFaucet, 1);
+    await firstValueFrom(wallet1.wallet.balance.utxo.total$.pipe(filter(({ coins }) => coins >= amountFromFaucet)));
+
+    const initialState = await getWalletStateSnapshot(wallet1.wallet);
+
+    const activePools = await wallet1.providers.stakePoolProvider.queryStakePools({
+      filters: { status: [Cardano.StakePoolStatus.Active] },
+      pagination: { limit: 2, startAt: 0 }
+    });
+    const poolId = activePools.pageResults[0].id;
+    expect(poolId).toBeDefined();
+    logger.debug(`Wallet funds will be staked to pool ${poolId}.`);
+    const certificates = createDelegationCertificates(initialState, poolId);
+
+    // Act
+
+    // Create stake tx
+    const unsignedTx = await wallet1.wallet.initializeTx({ certificates });
+    const signedTx = await wallet1.wallet.finalizeTx({ tx: unsignedTx });
+    await wallet1.wallet.submitTx(signedTx);
+
+    let currentEpoch = await firstValueFrom(wallet1.wallet.currentEpoch$);
+    logger.debug(`current epoch - ${currentEpoch.epochNo}`);
+    currentEpoch = await waitForEpoch(1);
+    logger.debug(`waited for epoch - ${currentEpoch.epochNo} to start txs `);
+    // workload for generating fees/rewards
+    await generateTxs(wallet1, wallet2);
+
+    // Assert
+
+    currentEpoch = await waitForEpoch(2);
```

Comment:
Maybe the 2nd waitForEpoch should be relative to when the delegation was made (not since 'now' which can be further ahead)

Reply:
Hi @mkazlauskas , I've changed `waitForEpoch` in `waitForNextEpoch` (not yet pushed); that's enough to perform the test and it makes the test much faster.
Was this the point of your suggestion or was you pointing to another problem I didn't saw?
Thank you